src/treecodes.py

```diff
@@ -1,238 +1,58 @@
-             'g2': {'group': 'T', 'code': 'g2'},
-             'g2-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'g2-osx-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'dromaeojs': {'group': 'T', 'code': 'd'},
-             'dromaeojs-e10s': {'group': 'T-e10s', 'code': 'd'},
-             'chromez': {'group': 'T', 'code': 'c'},
-             'chromez-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'chromez-osx-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'xperf': {'group': 'T', 'code': 'x'},
-             'xperf-e10s': {'group': 'T-e10s', 'code': 'x'},
-             'androidx86-set-4': {'group': 'S', 'code': '4'}}
+import requests
 
 
-def getGroup(name):
-    try:
-        group = name.split('-')[0]
-    except:
-        group = ''
-    return group
+class Treecodes:
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        url = 'https://treeherder.mozilla.org/api/project/%s/runnable_jobs/?format=json'
+        data = requests.get(url % repo_name).json()
+        joblist = data['results']
+        # skipping pgo - We run this infrequent enough that we should have all pgo results tested
+        self.joblist = [job for job in joblist if job['platform_option'] != 'pgo']
+        if len(self.joblist) > 0:
+            for job in self.joblist:
+                # the format of buildbot job is different from taskcluter jbo
+                if job['build_system_type'] == 'buildbot':
+                    testtype = job['ref_data_name'].split(' ')[-1]
+                    job_group_symbol = job['job_group_symbol'] \
+                        if job['job_group_symbol'] != '?' else ''
+                    job_type_symbol = job['job_type_symbol'] if job['job_type_symbol'] else ''
+                    self.tbplnames.update({testtype: {'group': job_group_symbol,
+                                                      'code': job_type_symbol}})
 
+    # return all jobtypes avaliable on buildbot for now
+    def jobtype_query(self):
+        jobtypes = []
+        for job in self.joblist:
+            testtype = job['ref_data_name'].split(' ')[-1]
+            if testtype in ['mochitest-jetpack', 'dep', 'nightly', 'non-unified',
+                            'valgrind', 'build']:
+                continue
+            else:
+                platform = job['platform']
+                buildtype = job['platform_option']
+                jobtypes.append([platform, buildtype, testtype])
 
-def getGroupCode(name):
-    try:
-        # strip useless blank in test name.
-        # e.g change ' mochitest-devtools-chrome-e10s-2' to
-        # 'mochitest-devtools-chrome-e10s-2'
-        code = tbplnames[name.strip()]['group']
-    except:
-        code = ''
-    return code
+        return jobtypes
 
+    def getgroup(self, name):
+        try:
+            group = name.split('-')[0]
```

Comment:
Could you please add a comment of in which cases we get to the except clause?

FYI, remember that some of these calculations you could do in the contrsuctor and have it all ready to inspect.